### PR TITLE
Switch to the @mymonero/mymonero-monero-client API

### PR DESCRIPTION
**Pull request by swansontec**
This PR builds, but needs more testing.

### CODE REVIEW

**Review by peachbits — APPROVED** [review]

File: android/src/main/cpp/jni.cpp [hunk 1]
```diff
@@ -17,19 +17,38 @@ Java_app_edge_reactnative_mymonerocore_MyMoneroModule_callMyMoneroJNI(
...
   jstring method,
-  jstring arguments
+  jobjectArray arguments
 ) {
   const std::string methodString = unpackJstring(env, method);
-  const std::string argumentsString = unpackJstring(env, arguments);
+
+  // Re-package the arguments:
+  std::vector<const std::string> strings;
+  jsize length = env->GetArrayLength(arguments);
```
Comment (swansontec) [inline android/src/main/cpp/jni.cpp:25]: Call `reserve` on the vector.

File: ios/MyMoneroCore.mm [hunk 2]
```diff
@@ -10,25 +10,46 @@ + (BOOL)requiresMainQueueSetup { return NO; }
...
-  arguments:(NSString *)arguments
+  arguments:(NSArray *)arguments
   resolver:(RCTPromiseResolveBlock)resolve
   rejecter:(RCTPromiseRejectBlock)reject
 ) {
   const std::string methodString = [method UTF8String];
-  const std::string argumentsString = [arguments UTF8String];
+
+  // Re-package the arguments:
+  std::vector<const std::string> strings;
```
Comment (swansontec) [inline ios/MyMoneroCore.mm]: Also call `reserve` here.

File: android/src/main/cpp/jni.cpp [hunk 3]
```diff
@@ -17,19 +17,38 @@ Java_app_edge_reactnative_mymonerocore_MyMoneroModule_callMyMoneroJNI(
...
 
   // Find the named method:
   for (int i = 0; i < myMoneroMethodCount; ++i) {
     if (myMoneroMethods[i].name != methodString) continue;
 
+    // Validate the argument count:
+    if (strings.size() != myMoneroMethods[i].argc) {
+      env->ThrowNew(
+        env->FindClass("java/lang/Exception"),
+        "mymonero-core-cpp incorrect argument count"
```
Comment (swansontec) [inline android/src/main/cpp/jni.cpp]: Replace "mymonero-core-cpp" messages with the right name.

File: src/index.d.ts [hunk 4]
```diff
@@ -1,7 +1,10 @@
...
  * The shape of the native C++ module exposed to React Native.
+ *
+ * You do not normally need this, but it is accessible as
+ * `require('react-native').NativeModules.MyMoneroCore`
  */
-export interface MyMoneroCore {
+export interface NativeMyMoneroCore {
   readonly callMyMonero: (
     name: string,
     jsonArguments: string
```
Comment (swansontec) [inline src/index.d.ts:10]: jsonArguments is string[] now.

**Review by swansontec — COMMENTED** [review]
Found a few things on final review.

**Pull request merged**